launcher/ui/dialogs/ModDownloadDialog.cpp

```diff
@@ -83,16 +115,38 @@ QList<BasePage *> ModDownloadDialog::getPages()
     };
 }
 
-void ModDownloadDialog::setSuggestedMod(const QString& name, ModDownloadTask* task)
+void ModDownloadDialog::addSelectedMod(const QString& name, ModDownloadTask* task)
+{
+    if(modTask.contains(name))
+        delete modTask.find(name).value();
+
+    if(task)
+        modTask.insert(name, task);
+    else
+        modTask.remove(name);
+
+    m_buttons->button(QDialogButtonBox::Ok)->setEnabled(!modTask.isEmpty());
+}
+
+void ModDownloadDialog::removeSelectedMod(const QString &name)
+{
+    if(modTask.contains(name))
+        delete modTask.find(name).value();
+    modTask.remove(name);
```

Comment:
nit: Maybe condense this down to `delete modTask.take(name)`?

Reply:
I think, since this is already such a small function, that leaving that as it is now would be fine, because `take()` will create a default-constructed `Task* ` just to be immediately deleted, whereas in this implementation it avoids the allocation altogether...

Reply:
Okay. Wasn't quite sure here, but sounds good then